scripts/secure-boot/generate-images/generate-image.sh

```diff
@@ -140,29 +143,35 @@ create_expired_cert() {
 }
 
 sign_img_and_create_iso() {
+    local _object_to_mount
     # sign hello.efi
     cp "$SCRIPTDIR"/hello.efi .
     error_check "Cannot copy hello.efi"
     if [ "$TESTNAME" != "BAD_FORMAT" ] && [ "$TESTNAME" != "NOT_SIGNED" ]; then
         sbsign --key cert.key --cert cert.crt --output signed-hello.efi hello.efi
         error_check "Cannot sign hello.efi"
     fi
-    # copy all things to the image
-    udisksctl loop-setup -f image.img
-    error_check "Cannot run udisksctl to create iso"
-    while [ ! -d "/run/media/$(whoami)/$IMAGELABEL" ]; do
+    # Copy all things to the image.
+    # Extract loop* from udisksctl output:
+    _object_to_mount="$(udisksctl loop-setup -f image.img | grep -o 'loop[0-9]\+' | sed 's/\.$//')"
+    error_check "Cannot run udisksctl to create ISO"
+    MOUNT_POINT=$(mount | grep $IMAGELABEL | awk '{print $3}')
+    while [ ! -d "$MOUNT_POINT" ]; do
         echo "Mounting $IMAGELABEL..."
+        udisksctl mount -p "block_devices/$_object_to_mount"
+        error_check "Cannot mount created ISO"
         sleep 0.2
+        MOUNT_POINT=$(mount | grep $IMAGELABEL | awk '{print $3}')
     done
     # copy everything to the image (except for the image itself)
     for file in *; do
 
         if [ "$file" != "image.img" ]; then
-            cp $file "/run/media/$(whoami)/$IMAGELABEL/"
+            cp $file "$MOUNT_POINT/"
         fi
 
     done
-    umount "/run/media/$(whoami)/$IMAGELABEL"
+    umount "$MOUNT_POINT"
```

Comment:
We should probably remove the loop device after it is used, so that we don't leave it hanging after it is done, something like this works just fine:

```
umount "$MOUNT_POINT"
sudo losetup -d /dev/$_object_to_mount
```

Besides that LGTM and works as expected.

Reply:
Addressed here 48eaae863f110335d81b2818fa14472826990423.

Reply:
Loop-device and mount points will remain if script fails somewhere earlier and exits. `trap` would be probably better.